pyensembl/gene.py

```diff
@@ -55,21 +56,20 @@ def transcripts(self):
         Property which dynamically construct transcript objects for all
         transcript IDs associated with this gene.
         """
-        transcript_ids_query = """
-            SELECT transcript_id
-            FROM ensembl
-            WHERE gene_id = ?
-            AND feature = 'transcript'
-        """
-        cursor = self.db.execute(transcript_ids_query, [self.id])
-        results = cursor.fetchall()
+        transcript_id_results = self.db.query(
+            select_column_names=['transcript_id'],
+            filter_column='gene_id',
+            filter_value=self.id,
+            feature='transcript',
+            distinct=False,
+            required=False)
 
         # We're doing a SQL query for each transcript ID to fetch
         # its particular information, might be more efficient if we
         # just get all the columns here, but how do we keep that modular?
         return [
-            Transcript(result[0], self.db)
-            for result in results
+            Transcript(result[0], self.db, self.reference)
```

Comment:
Each `GeneEach transcript object can (lazily) fetch its sequence from a `ReferenceTranscripts` object, so `Transcript` needs to have